src/Compilers/CSharp/Portable/Symbols/TypeSymbolExtensions.cs

```diff
@@ -1203,6 +1213,29 @@ internal static int ComputeHashCode(this NamedTypeSymbol type)
                 code++;
             }
             return code;
+
+            static bool wasConstructedForAnnotations(NamedTypeSymbol type)
+            {
+                do
+                {
+                    var typeArguments = type.TypeArgumentsWithAnnotationsNoUseSiteDiagnostics;
+                    var typeParameters = type.ConstructedFrom.TypeParameters;
+
+                    for (int i = 0; i < typeArguments.Length; i++)
+                    {
+                        if (!typeArguments[i].Type.OriginalDefinition.Equals(typeParameters[i], TypeCompareKind.CLRSignatureCompareOptions))
```

Comment:
>TypeCompareKind.CLRSignatureCompareOptions [](start = 96, length = 42)

Passing ```TypeCompareKind``` here is not necessary, we are comparing type parameter symbols.

Reply:
That is assuming that both sides are definitions

---
In reply to: [283573830](https://github.com/dotnet/roslyn/pull/35613#discussion_r283573830) [](ancestors = 283573830)